tools/applicationinsights-web-snippet/src/snippet.ts

```diff
@@ -138,35 +181,114 @@ declare var cfg:ISnippetConfig;
                 + "Z";
         }
 
-        function _createEnvelope(iKey:string, theType:string) {
-            let tags = {};
-            let type = "Browser";
-            tags[strAiDevice + "id"] = type[strToLowerCase]();
-            tags[strAiDevice + "type"] = type;
-            tags[strAiOperationName] = locn && locn.pathname || "_unknown_";
-            tags[strAiSdkVersion] = "javascript:snippet_" + (appInsights.sv || appInsights.version);
-
-            let envelope:IEnvelope = {
-                time: _getTime(),
-                iKey: iKey,
-                name: "Microsoft.ApplicationInsights." + iKey.replace(/-/g, strEmpty) + "." + theType,
-                sampleRate: 100,
-                tags: tags,
-                data: {
-                    baseData: {
-                        ver: 2
-                    }
-                },
-                ver: undefined,
-                seq: "1",
-                aiDataContract: undefined
+        function _addTimeZone(envelope: any): void {
+            // Add time zone
+            const timeZone: number = new Date().getTimezoneOffset();
+            let minutes: number = timeZone % 60;
+            let hours: number = (timeZone - minutes) / 60;
+            let timeZonePrefix: string = "+";
+            if (hours > 0) {
+                timeZonePrefix = "-";
+            }
+            hours = Math.abs(hours);
+            minutes = Math.abs(minutes);
+        
+            envelope.ext = envelope.ext || {};
+            envelope.ext.loc = {  // Add time zone
+                tz: timeZonePrefix + (hours < 10 ? "0" + hours : hours.toString()) + ":" + (minutes < 10 ? "0" + minutes : minutes.toString())
             };
+        }
 
-            return envelope;
+        function _getTenantId(apiKey: string): string {
+            let result: string = "";
+        
+            if (apiKey) {
+                const indexTenantId: number = apiKey.indexOf("-");
+                if (indexTenantId > -1) {
+                    result = apiKey.substring(0, indexTenantId);
+                }
+            }
+            return result;
+        }
+        
+        function _addUser(envelope: any): void {
+            const strUndefined: string = "undefined";
+            // Add user language
+            if (typeof navigator !== strUndefined) {
+                const nav: Navigator & { userLanguage?: string } = navigator;
+                envelope.ext = envelope.ext || {};
+                envelope.ext.user = {
+                    locale: nav.userLanguage || nav.language
+                };
+            }
+        }
+
+        function _createEnvelope(iKey:string, theType:string) {
```

Comment:
I know this suggestion is probably going to make the resulting file larger, but I'm thinking maybe we should extract the creation of the reporting "error" (envelope) to a new file
So
`_createAiEnvelope()` and `_createOneDsEnvelope()`

So if we keep the code on lines 306..310 with the generation stuff you have added this "should" still cause the minification to work nicely and should make this core file a little easier to read...

We "may" even just have the unique portions of AI / 1DS in each of those files (now that we know we have these differences we need to do ☹️ )

Reply:
A bit like the way you have the 1dsType.ts, maybe we have a 1dsSupport.ts type thing